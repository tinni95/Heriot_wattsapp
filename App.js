import React from "react";
import { NativeRouter } from "react-router-native";
import { SafeAreaView, ScrollView, View, StyleSheet, Text } from "react-native";
import { SearchBar } from "react-native-elements";

import {
    main,
    secondary,
    searchBar,
} from "./client-configs/stefanos/constants/colors";

import { categories } from "./dummy";
import NavigationBar from "./components/navigationBar";
import InfoBar from "./components/infoBar";
import Menu from "./components/menu";

export default class Appe extends React.Component {
    state = {
        search: "",
    };

    updateSearch = search => {
        this.setState({ search });
    };

    render() {
        const { search } = this.state;
        return (
            <NativeRouter>
                <ScrollView>
                    <InfoBar style={styles.infoBar} />
                    <View style={styles.mainHeaders}>
                        <View style={styles.logoContainer}>
                            <Text style={styles.logo}>Logo</Text>
                        </View>
                        <NavigationBar {...categories} />
                    </View>
                    <View style={styles.mainConent}>
                        <SearchBar
                            containerStyle={styles.searchBarContainer}
                            inputContainerStyle={styles.searchBar}
                            placeholder={"Type something..."}
                            onChangeText={this.updateSearch}
                            value={search}
                            placeholderTextColor={"white"}
                            color={"white"}
                        />
                        <Menu {...categories} />
                    </View>
                </ScrollView>
            </NativeRouter>
        );
    }
}

const styles = StyleSheet.create({
    appContainer: {
        flex: 1,
        flexDirection: "column",
        alignItems: "stretch",
    },
    infoBar: {
        backgroundColor: secondary,
        paddingBottom: 15,
    },
    mainHeaders: {
        backgroundColor: main,
    },
    mainConent: {},
    searchBarContainer: {
        backgroundColor: "white",
        borderBottomColor: "transparent",
        borderTopColor: "transparent",
    },
    searchBar: {
        backgroundColor: secondary,
        borderRadius: 10,
    },
    logoContainer: {
        margin: 20,
    },
    logo: {
        color: "white",
        fontSize: 50,
        textAlign: "center",
    },
});
