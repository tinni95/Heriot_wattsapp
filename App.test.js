import App, { updateSearch } from "./App";
import renderer from "react-test-renderer";
import React from "react";
import { SearchBar } from "react-native-elements";

describe("app", () => {
    let rendered;
    beforeEach(() => {
        rendered = renderer.create(<App />);
    });

    it("renders", () => {
        expect(rendered).toBeTruthy();
    });

    it("it matches Snapshot", () => {
        const snapshottable = rendered.toJSON();
        expect(snapshottable).toMatchSnapshot();
    });
});
