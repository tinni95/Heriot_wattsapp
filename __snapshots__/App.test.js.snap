// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app it matches Snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#E0A749",
        "paddingBottom": 15,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingLeft": 8,
          "paddingTop": 40,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginRight": 20,
          }
        }
      >
        <View
          style={
            Object {
              "marginRight": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
            underlayColor="white"
          >
            <Text />
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
            }
          }
        >
          01314471135
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginRight": 20,
          }
        }
      >
        <View
          style={
            Object {
              "marginRight": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
            underlayColor="white"
          >
            <Text />
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
            }
          }
        >
          open:11am-11pm
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#301D5C",
      }
    }
  >
    <View
      style={
        Object {
          "margin": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 50,
            "textAlign": "center",
          }
        }
      >
        Logo
      </Text>
    </View>
    <RCTScrollView
      horizontal={true}
      scrollEventThrottle={16}
      style={
        Object {
          "height": 100,
          "marginTop": 20,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "width": 330,
            }
          }
        />
        <View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "opacity": 1,
                "padding": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "#E0A749",
                  "borderBottomWidth": 2,
                  "height": 40,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#D6D1D1",
                      "fontSize": 22,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "white",
                      "fontSize": 25,
                      "paddingTop": -10,
                    },
                  ]
                }
              >
                Mains
              </Text>
            </View>
          </View>
        </View>
        <View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "opacity": 1,
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#D6D1D1",
                  "fontSize": 22,
                  "textAlign": "center",
                }
              }
            >
              Starters
            </Text>
          </View>
        </View>
        <View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "opacity": 1,
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#D6D1D1",
                  "fontSize": 22,
                  "textAlign": "center",
                }
              }
            >
              Desserts
            </Text>
          </View>
        </View>
        <View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "opacity": 1,
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#D6D1D1",
                  "fontSize": 22,
                  "textAlign": "center",
                }
              }
            >
              Special
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "width": 330,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>,
  <RCTScrollView>
    <View>
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "transparent",
              "borderBottomWidth": 1,
              "borderTopColor": "transparent",
              "borderTopWidth": 1,
              "padding": 8,
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 0,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#E0A749",
                  "borderBottomWidth": 0,
                  "borderColor": "#86939e",
                  "borderRadius": 10,
                  "flexDirection": "row",
                  "minHeight": 30,
                  "overflow": "hidden",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginLeft": 8,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                    theme={
                      Object {
                        "colors": Object {
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "primary": "#2196f3",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "primary": "#007aff",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#8F0CE8",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      }
                    }
                    underlayColor="white"
                  >
                    <Text />
                  </View>
                </View>
              </View>
              <TextInput
                allowFontScaling={true}
                color="white"
                onBlur={[Function]}
                onChangeText={[Function]}
                onClear={[Function]}
                onFocus={[Function]}
                placeholder="Type something..."
                placeholderTextColor="white"
                platform="default"
                rejectResponderTermination={true}
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#fff",
                    "flex": 1,
                    "fontSize": 18,
                    "marginLeft": 10,
                    "minHeight": 40,
                  }
                }
                testID="searchInput"
                underlineColorAndroid="transparent"
                value=""
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "marginRight": 8,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "marginTop": 20,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#F4F4F4",
                  "borderRadius": 10,
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 100,
                  "margin": 8,
                  "width": 320,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 6,
                    "flexDirection": "column",
                    "margin": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 24,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mangia
                </Text>
                <Text
                  style={
                    Object {
                      "fontSize": 16,
                      "marginTop": 15,
                    }
                  }
                >
                  £
                  20
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "backgroundColor": "#348653",
                    "borderBottomRightRadius": 10,
                    "borderTopRightRadius": 10,
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 20,
                      "textAlign": "center",
                    }
                  }
                >
                  +
                </Text>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "marginTop": 20,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#F4F4F4",
                  "borderRadius": 10,
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 100,
                  "margin": 8,
                  "width": 320,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 6,
                    "flexDirection": "column",
                    "margin": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 24,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mangiaa
                </Text>
                <Text
                  style={
                    Object {
                      "fontSize": 16,
                      "marginTop": 15,
                    }
                  }
                >
                  £
                  10
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "backgroundColor": "#348653",
                    "borderBottomRightRadius": 10,
                    "borderTopRightRadius": 10,
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 20,
                      "textAlign": "center",
                    }
                  }
                >
                  +
                </Text>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "marginTop": 20,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#F4F4F4",
                  "borderRadius": 10,
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 100,
                  "margin": 8,
                  "width": 320,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 6,
                    "flexDirection": "column",
                    "margin": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 24,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mangiia
                </Text>
                <Text
                  style={
                    Object {
                      "fontSize": 16,
                      "marginTop": 15,
                    }
                  }
                >
                  £
                  10
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "backgroundColor": "#348653",
                    "borderBottomRightRadius": 10,
                    "borderTopRightRadius": 10,
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 20,
                      "textAlign": "center",
                    }
                  }
                >
                  +
                </Text>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "marginTop": 20,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#F4F4F4",
                  "borderRadius": 10,
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 100,
                  "margin": 8,
                  "width": 320,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 6,
                    "flexDirection": "column",
                    "margin": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 24,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mangi2a
                </Text>
                <Text
                  style={
                    Object {
                      "fontSize": 16,
                      "marginTop": 15,
                    }
                  }
                >
                  £
                  10
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "backgroundColor": "#348653",
                    "borderBottomRightRadius": 10,
                    "borderTopRightRadius": 10,
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 20,
                      "textAlign": "center",
                    }
                  }
                >
                  +
                </Text>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "marginTop": 20,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#F4F4F4",
                  "borderRadius": 10,
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 100,
                  "margin": 8,
                  "width": 320,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 6,
                    "flexDirection": "column",
                    "margin": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 24,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mangi23a
                </Text>
                <Text
                  style={
                    Object {
                      "fontSize": 16,
                      "marginTop": 15,
                    }
                  }
                >
                  £
                  10
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "backgroundColor": "#348653",
                    "borderBottomRightRadius": 10,
                    "borderTopRightRadius": 10,
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 20,
                      "textAlign": "center",
                    }
                  }
                >
                  +
                </Text>
              </View>
            </View>
          </View>
          <View>
            <Text
              style={
                Object {
                  "fontSize": 25,
                  "fontWeight": "bold",
                  "margin": 25,
                }
              }
            >
              Panini
            </Text>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "marginTop": 20,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F4F4F4",
                    "borderRadius": 10,
                    "flex": 1,
                    "flexDirection": "row",
                    "height": 100,
                    "margin": 8,
                    "width": 320,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 6,
                      "flexDirection": "column",
                      "margin": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 24,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Panini With Ham
                  </Text>
                  <Text
                    style={
                      Object {
                        "fontSize": 16,
                        "marginTop": 15,
                      }
                    }
                  >
                    £
                    10
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignContent": "center",
                      "backgroundColor": "#348653",
                      "borderBottomRightRadius": 10,
                      "borderTopRightRadius": 10,
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 20,
                        "textAlign": "center",
                      }
                    }
                  >
                    +
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
